### PR TITLE
fix: docker compose yml

### PR DESCRIPTION
**Pull request by Emrehzl94**
Docker compose file updated from quick start documentation.

### CODE REVIEW

File: kafka-connect-neo4j/doc/docker-compose.yml [hunk 1]
```diff
@@ -1,86 +1,83 @@
...
     environment:
-      SCHEMA_REGISTRY_HOST_NAME: schema_registry
-      SCHEMA_REGISTRY_KAFKASTORE_CONNECTION_URL: 'zookeeper:2181'
+      SCHEMA_REGISTRY_HOST_NAME: schema-registry
+      SCHEMA_REGISTRY_KAFKASTORE_BOOTSTRAP_SERVERS: 'broker:29092'
+      SCHEMA_REGISTRY_LISTENERS: http://0.0.0.0:8081
 
   connect:
-    image: confluentinc/cp-kafka-connect
+    image: cnfldemos/cp-server-connect-datagen:0.6.0-7.3.0
```
Comment (fbiville) [inline kafka-connect-neo4j/doc/docker-compose.yml:68]: is this the right image?

Comment (Emrehzl94) [reply kafka-connect-neo4j/doc/docker-compose.yml:68]: yeah, doesn't look so but I also don't know why this is the image in the documents

Comment (ali-ince) [reply kafka-connect-neo4j/doc/docker-compose.yml]: I think we can keep it as `confluentinc/cp-kafka-connect:7.3.0`.

Comment (Emrehzl94) [reply kafka-connect-neo4j/doc/docker-compose.yml]: There is an error in connect instance logs like 
```
2024-07-19 09:12:39 [2024-07-19 07:12:39,449] ERROR Uncaught exception in REST call to /v1/metadata/id (org.apache.kafka.connect.runtime.rest.errors.ConnectExceptionMapper)
2024-07-19 09:12:39 javax.ws.rs.NotFoundException: HTTP 404 Not Found
```
when I tried `confluentinc/cp-kafka-connect:7.3.0` 

but `confluentinc/cp-server-connect:7.3.0` seems to work seamlessly

**Review by fbiville — APPROVED** [review]

**Pull request merged**